test/fuzz/lua/mvcc_test.lua

```diff
@@ -0,0 +1,778 @@
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = true}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = true}}),
+            create_hash_idx_meta({{is_nullable = false},
+                                  {is_nullable = false}}),
+        }
+    },
+    {
+        name = 'vinyl',
+        space = 'v',
+        idxs = {
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = false}}),
+            create_tree_idx_meta({{is_nullable = false},
+                                  {is_nullable = true}}),
+            create_tree_idx_meta({{is_nullable = true},
+                                  {is_nullable = true}}),
+        }
+    },
+}
+
+local function generate_txn_repro_stmt(stmt)
+    return ("tx%d('%s') -- %s\n"):format(stmt.tid, stmt.str,
+                                         json.encode(stmt.res))
+end
+
+local function generate_txn_serial_stmt(stmt)
+    return ("%s -- tx%d: %s\n"):format(stmt.str, stmt.tid,
+                                       json.encode(stmt.res))
+end
+
+local INIT_CODE =
+[[os.execute('rm -rf *.snap *.xlog *.vylog 512 513')
+
+local ffi = require('ffi')
+local fiber = require('fiber')
+local json = require('json')
+local log = require('log')
+local txn_proxy = require('test.box.lua.txn_proxy')
+
+box.cfg{memtx_use_mvcc_engine = %s}
+fiber.set_max_slice({warn = 6000, err = 6000})
+
+]]
+
+local function generate_init_code(memtx_use_mvcc_engine)
+    return (INIT_CODE):format(memtx_use_mvcc_engine and 'true' or 'false')
+end
+
+local SCHEMA_CODE =
+[[box.schema.space.create('m', {engine = 'memtx'})
+box.space.m:create_index('p', {parts = {
+    {1, 'uint'},
+    {2, 'uint'}}})
+box.space.m:create_index('s1', {parts = {
+    {3, 'uint', exclude_null = true},
+    {2, 'uint'}}})
+box.space.m:create_index('s2', {parts = {
+    {1, 'uint'},
+    {4, 'uint', exclude_null = true}}})
+box.space.m:create_index('s3', {parts = {
+    {3, 'uint', exclude_null = true},
+    {4, 'uint', exclude_null = true}}})
+box.space.m:create_index('s4', {type = 'HASH', parts = {
+    {1, 'uint'},
+    {2, 'uint'}}})
+box.schema.space.create('v', {engine = 'vinyl'})
+box.space.v:create_index('p', {parts = {
+    {1, 'uint'},
+    {2, 'uint'}}})
+box.space.v:create_index('s1', {parts = {
+    {3, 'uint', exclude_null = true},
+    {2, 'uint'}}})
+box.space.v:create_index('s2', {parts = {
+    {1, 'uint'},
+    {4, 'uint', exclude_null = true}}})
+box.space.v:create_index('s3', {parts = {
+    {3, 'uint', exclude_null = true},
+    {4, 'uint', exclude_null = true}}})
+
```

Comment:
Let's add a functional index for `memtx`. It seems to be easy when scheme is pre-defined, not randomly generated. We can do it under an `--enable_func_index` option.